SIGSEGV

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x95e28d]

goroutine 3244 [running]:
github.com/redis/rueidis.(*lru).purge(0xc0027e1200, {0xc0cb1c53c8, 0x13}, 0xc04b7fbba0)
	/home/runner/go/pkg/mod/github.com/redis/rueidis@v1.0.33/lru.go:299 +0xad
github.com/redis/rueidis.(*lru).Delete(0xc0027e1200, {0xc0dca2c340, 0x1, 0x0?})
	/home/runner/go/pkg/mod/github.com/redis/rueidis@v1.0.33/lru.go:318 +0x15f
github.com/redis/rueidis.(*pipe).handlePush(0xc000a27440, {0xc0afeaf300, 0x0?, 0xc0afeaf300?})
	/home/runner/go/pkg/mod/github.com/redis/rueidis@v1.0.33/pipe.go:640 +0x4fa
github.com/redis/rueidis.(*pipe)._backgroundRead(0xc000a27440)
	/home/runner/go/pkg/mod/github.com/redis/rueidis@v1.0.33/pipe.go:487 +0x32a
github.com/redis/rueidis.(*pipe)._background(0xc000a27440)
	/home/runner/go/pkg/mod/github.com/redis/rueidis@v1.0.33/pipe.go:342 +0x88
created by github.com/redis/rueidis.(*pipe).background.func1 in goroutine 3134
	/home/runner/go/pkg/mod/github.com/redis/rueidis@v1.0.33/pipe.go:325 +0x56
```

--- a/lru.go
+++ b/lru.go
@@ -78,7 +78,7 @@ func (c *lru) Flight(key, cmd string, ttl time.Duration, now time.Time) (v Redis
 
 	c.mu.RLock()
 	if kc, ok = c.store[key]; ok {
-		if ele, ok = kc.cache[cmd]; ok {
+		if ele = kc.cache[cmd]; ele != nil {
 			e = ele.Value.(*cacheEntry)
 			v = e.val
 			back = c.list.Back()
@@ -109,7 +109,7 @@ func (c *lru) Flight(key, cmd string, ttl time.Duration, now time.Time) (v Redis
 		kc = &keyCache{cache: make(map[string]*list.Element, 1), key: key}
 		c.store[key] = kc
 	}
-	if ele, ok = kc.cache[cmd]; ok {
+	if ele = kc.cache[cmd]; ele != nil {
 		if e = ele.Value.(*cacheEntry); e.val.typ == 0 || e.val.relativePTTL(now) > 0 {
 			atomic.AddUint32(&kc.hits, 1)
 			v = e.val
@@ -123,8 +123,7 @@ func (c *lru) Flight(key, cmd string, ttl time.Duration, now time.Time) (v Redis
 	}
 	atomic.AddUint32(&kc.miss, 1)
 	v.setExpireAt(now.Add(ttl).UnixMilli())
-	c.list.PushBack(&cacheEntry{cmd: cmd, kc: kc, val: v, ch: make(chan struct{})})
-	kc.cache[cmd] = c.list.Back()
+	kc.cache[cmd] = c.list.PushBack(&cacheEntry{cmd: cmd, kc: kc, val: v, ch: make(chan struct{})})
 ret:
 	c.mu.Unlock()
 	return v, ce
@@ -137,7 +136,7 @@ func (c *lru) Flights(now time.Time, multi []CacheableTTL, results []RedisResult
 	for i, ct := range multi {
 		key, cmd := cmds.CacheKey(ct.Cmd)
 		if kc, ok := c.store[key]; ok {
-			if ele, ok := kc.cache[cmd]; ok {
+			if ele := kc.cache[cmd]; ele != nil {
 				e := ele.Value.(*cacheEntry)
 				v := e.val
 				if v.typ == 0 {
@@ -191,7 +190,7 @@ func (c *lru) Flights(now time.Time, multi []CacheableTTL, results []RedisResult
 			kc = &keyCache{cache: make(map[string]*list.Element, 1), key: key}
 			c.store[key] = kc
 		}
-		if ele, ok := kc.cache[cmd]; ok {
+		if ele := kc.cache[cmd]; ele != nil {
 			e := ele.Value.(*cacheEntry)
 			v := e.val
 			if v.typ == 0 {
@@ -211,8 +210,7 @@ func (c *lru) Flights(now time.Time, multi []CacheableTTL, results []RedisResult
 		atomic.AddUint32(&kc.miss, 1)
 		v := RedisMessage{}
 		v.setExpireAt(now.Add(multi[i].TTL).UnixMilli())
-		c.list.PushBack(&cacheEntry{cmd: cmd, kc: kc, val: v, ch: make(chan struct{})})
-		kc.cache[cmd] = c.list.Back()
+		kc.cache[cmd] = c.list.PushBack(&cacheEntry{cmd: cmd, kc: kc, val: v, ch: make(chan struct{})})
 		missed[j] = i
 		j++
 	}
@@ -224,7 +222,7 @@ func (c *lru) Update(key, cmd string, value RedisMessage) (pxat int64) {
 	var ch chan struct{}
 	c.mu.Lock()
 	if kc, ok := c.store[key]; ok {
-		if ele, ok := kc.cache[cmd]; ok {
+		if ele := kc.cache[cmd]; ele != nil {
 			if e := ele.Value.(*cacheEntry); e.val.typ == 0 {
 				pxat = value.getExpireAt()
 				cpttl := e.val.getExpireAt()
@@ -264,7 +262,7 @@ func (c *lru) Cancel(key, cmd string, err error) {
 	var ch chan struct{}
 	c.mu.Lock()
 	if kc, ok := c.store[key]; ok {
-		if ele, ok := kc.cache[cmd]; ok {
+		if ele := kc.cache[cmd]; ele != nil {
 			if e := ele.Value.(*cacheEntry); e.val.typ == 0 {
 				e.err = err
 				ch = e.ch
@@ -296,12 +294,16 @@ func (c *lru) GetTTL(key, cmd string) (ttl time.Duration) {
 func (c *lru) purge(key string, kc *keyCache) {
 	if kc != nil {
 		for cmd, ele := range kc.cache {
-			if e := ele.Value.(*cacheEntry); e.val.typ != 0 { // do not delete pending entries
-				if delete(kc.cache, cmd); len(kc.cache) == 0 {
-					delete(c.store, key)
+			if ele != nil {
+				e := ele.Value.(*cacheEntry)
+				if e.val.typ == 0 { // do not delete pending entries
+					continue
 				}
 				c.list.Remove(ele)
 				c.size -= e.size
+			}
+			if delete(kc.cache, cmd); len(kc.cache) == 0 {
+				delete(c.store, key)
 			}
 		}
 	}
@@ -325,9 +327,11 @@ func (c *lru) Close(err error) {
 	c.mu.Lock()
 	for _, kc := range c.store {
 		for _, ele := range kc.cache {
-			if e := ele.Value.(*cacheEntry); e.val.typ == 0 {
-				e.err = err
-				close(e.ch)
+			if ele != nil {
+				if e := ele.Value.(*cacheEntry); e.val.typ == 0 {
+					e.err = err
+					close(e.ch)
+				}
 			}
 		}
 	}

